src/main/java/mate/academy/service/impl/CinemaHallServiceImpl.java

```diff
@@ -0,0 +1,31 @@
+package mate.academy.service.impl;
+
+import java.util.List;
+import java.util.NoSuchElementException;
+import mate.academy.dao.CinemaHallDao;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.CinemaHall;
+import mate.academy.service.CinemaHallService;
+
+@Service
+public class CinemaHallServiceImpl implements CinemaHallService {
+    @Inject
+    private CinemaHallDao cinemaHallDao;
+
+    @Override
+    public CinemaHall add(CinemaHall cinemaHall) {
+        return cinemaHallDao.add(cinemaHall);
+    }
+
+    @Override
+    public CinemaHall get(Long id) {
+        return cinemaHallDao.get(id)
+                .orElseThrow(() -> new NoSuchElementException("Can't get cinema hall"));
```

Comment:
let's also specify the ID in your exception message here. Fix the same issue in all places

Reply:
But isn't it a violation of security or a violation of encapsulation? I mean, aren't we throwing exceptions on the service layer to the end user? So he shouldn't know about redundant information for him, like the id of something?

Reply:
1. It depends of an specific application, for some entities id can be hidden for some not. E.g. you can often see example of the REST request get user by id, where the `id` is a part of the url parameter.
2. Throwing an exception here doesn't mean that exception message will be shown to the end user, message will be put to the application log, and should have contain all details for possible investigation the problem.